events/userCredit.js

```diff
@@ -6,25 +6,46 @@
 
 module.exports = async (user, amount) => {
   try {
-    let userProfile = await user.client.db.get(`SELECT bastionCurrencies FROM profiles WHERE userID=${user.id}`);
+    let userModel = await user.client.database.models.guildMember.findOne({
+      attributes: [ 'bastionCurrencies' ],
+      where: {
+        userID: user.id
+      }
+    });
 
     /*
-    * If the user doesn't have a profile, yet, we don't allow
-    * to deduct Bastion Currencies from them.
-    */
-    if (!userProfile) return;
+     * If the user doesn't have a profile, yet, we don't allow
+     * to deduct Bastion Currencies from them.
+     */
+    if (!userModel) return;
 
     /*
-    * Deduct the given amount of Bastion Currencies from the user's account.
-    * Yes, if they have less Bastion Currencies then the given amount,
-    * that will still be deducted from their account.
-    */
-    await user.client.db.run(`UPDATE profiles SET bastionCurrencies=${parseInt(userProfile.bastionCurrencies) - parseInt(amount)} WHERE userID=${user.id}`);
+     * Deduct the given amount of Bastion Currencies from the user's account.
+     * Yes, if they have less Bastion Currencies then the given amount,
+     * that will still be deducted from their account.
+     */
+    await user.client.database.models.guildMember.update({
+      bastionCurrencies: parseInt(userModel.dataValues.bastionCurrencies) - parseInt(amount)
+    },
+    {
+      where: {
+        userID: user.id
+      },
+      fields: [ 'bastionCurrencies' ]
+    });
 
     /*
-    * Add the transaction detail to transactions table.
-    */
-    await user.client.db.run('INSERT INTO transactions (userID, type, amount) VALUES (?, ?, ?)', [ user.id, 'userCredit', parseInt(amount) ]);
+     * Add the transaction detail to transactions table.
+     */
+    await user.client.database.models.transaction.create({
+      userID: user.id,
+      guildID: 'ID', // TODO: Add support for guild ID.
```

Comment:
This is done temporarily to imitate global transactions (which was available up to v6). After this is merged, I'll implement changes to support per server transaction records.